model_compression_toolkit/core/pytorch/back2framework/mixed_precision_model_builder.py

```diff
@@ -13,118 +13,261 @@
 # limitations under the License.
 # ==============================================================================
 
-from typing import List, Any, Tuple
+from typing import List, Any, Tuple, Union, Dict
 
 import torch
+from mct_quantizers import PytorchQuantizationWrapper, QuantizationTarget, \
+    PytorchActivationQuantizationHolder
+from mct_quantizers.common.constants import ACTIVATION_HOLDER_QUANTIZER
+from mct_quantizers.common.get_quantizers import get_inferable_quantizer_class
+from mct_quantizers.pytorch.quantizers import BasePyTorchInferableQuantizer
 
 from model_compression_toolkit.core import FrameworkInfo
 from model_compression_toolkit.core import common
 from model_compression_toolkit.core.common import BaseNode
-from model_compression_toolkit.core.common.graph.functional_node import FunctionalNode
 from model_compression_toolkit.core.common.user_info import UserInformation
-from model_compression_toolkit.core.pytorch.back2framework.instance_builder import node_builder
-from model_compression_toolkit.core.pytorch.back2framework.pytorch_model_builder import PyTorchModelBuilder, \
-    PytorchModel
+from model_compression_toolkit.core.pytorch.back2framework.pytorch_model_builder import PyTorchModelBuilder
 
 from model_compression_toolkit.core.pytorch.default_framework_info import DEFAULT_PYTORCH_INFO
-from model_compression_toolkit.core.pytorch.mixed_precision.mixed_precision_wrapper import PytorchMixedPrecisionWrapper
+from model_compression_toolkit.core.pytorch.mixed_precision.configurable_activation_quantizer import \
+    ConfigurableActivationQuantizer
+from model_compression_toolkit.core.pytorch.mixed_precision.configurable_weights_quantizer import \
+    ConfigurableWeightsQuantizer
 
+from model_compression_toolkit.exporter.model_wrapper.pytorch.builder.node_to_quantizer import \
+    get_weights_inferable_quantizer_kwargs, get_activation_inferable_quantizer_kwargs
+from model_compression_toolkit.logger import Logger
 
-class MixedPrecisionPyTorchModel(PytorchModel):
 
+class MixedPrecisionPyTorchModelBuilder(PyTorchModelBuilder):
+    """
+    Mixed-precision PyTorch model.
+    """
     def __init__(self,
                  graph: common.Graph,
-                 append2output=None):
+                 append2output=None,
+                 fw_info: FrameworkInfo = DEFAULT_PYTORCH_INFO,
```

Comment:
Maybe it's better to use directly DEFAULT_PYTORCH_INFO instead of passing it?

Reply:
I don't want to do that because it would require changing the call for the builder from pytorch_implementation which is a generic call that is used in other places as well (we can talk about this offline if it's not clear)